static/app/components/guidedSteps/guidedSteps.spec.tsx

```diff
@@ -1,99 +1,128 @@
+        stepKey="step-1"
+        title="Step 1 Title"
+        isCompleted={completedSteps?.includes('step-1')}
+      >
+        This is the first step.
+        <GuidedSteps.StepButtons />
+      </GuidedSteps.Step>
+      <GuidedSteps.Step
+        stepKey="step-2"
+        title="Step 2 Title"
+        isCompleted={completedSteps?.includes('step-2')}
+      >
+        This is the second step.
+        <GuidedSteps.StepButtons />
+      </GuidedSteps.Step>
+      <GuidedSteps.Step
+        stepKey="step-3"
+        title="Step 3 Title"
+        isCompleted={completedSteps?.includes('step-3')}
+      >
+        This is the third step.
+        <GuidedSteps.StepButtons />
+      </GuidedSteps.Step>
+      <GuidedSteps.Step
+        stepKey="step-4"
+        title="Step 4 Title"
+        isCompleted={completedSteps?.includes('step-4')}
+      >
+        This is the fourth step.
+        <GuidedSteps.StepButtons />
+      </GuidedSteps.Step>
+    </GuidedSteps>
+  );
+}
+
+describe('GuidedSteps', function () {
+  const router = RouterFixture();
+
+  it('renders step based on the query param and shows previous ones as completed', function () {
+    render(<Component />, {
+      router: {
+        ...router,
+        location: {
+          ...router.location,
+          query: {guidedStep: '3'},
+        },
+      },
+    });
 
     expect(screen.queryByText('This is the first step.')).not.toBeInTheDocument();
-    expect(screen.getByText('This is the second step.')).toBeInTheDocument();
-    expect(screen.queryByText('This is the third step.')).not.toBeInTheDocument();
+    expect(screen.queryByText('This is the second step.')).not.toBeInTheDocument();
+    expect(screen.getByText('This is the third step.')).toBeInTheDocument();
+    expect(screen.queryByText('This is the fourth step.')).not.toBeInTheDocument();
 
-    // First step is shown as completed
+    // Previous steps are shown as completed
     expect(
       within(screen.getByTestId('guided-step-1')).getByTestId('icon-check-mark')
     ).toBeInTheDocument();
+    expect(
+      within(screen.getByTestId('guided-step-2')).getByTestId('icon-check-mark')
+    ).toBeInTheDocument();
   });
 
-  it('starts at the first incomplete step', function () {
-    render(
-      <GuidedSteps>
-        <GuidedSteps.Step stepKey="step-1" title="Step 1 Title" isCompleted>
-          This is the first step.
-          <GuidedSteps.StepButtons />
-        </GuidedSteps.Step>
-        <GuidedSteps.Step stepKey="step-2" title="Step 2 Title" isCompleted={false}>
-          This is the second step.
-          <GuidedSteps.StepButtons />
-        </GuidedSteps.Step>
-        <GuidedSteps.Step stepKey="step-3" title="Step 3 Title" isCompleted={false}>
-          This is the third step.
-          <GuidedSteps.StepButtons />
-        </GuidedSteps.Step>
-      </GuidedSteps>
+  it('update url query param when navigating through steps', async function () {
+    render(<Component />, {
+      router,
```

Comment:
Oh, also! There's a new option that I haven't documented yet (so no need to make the change, just letting you know), that makes testing location changes easier: https://github.com/getsentry/sentry/pull/83400

Note that I recently changed the option from `disableRouterMocks: true` to `enableRouterMocks: false` since I'm in the process of making it the default behavior, but otherwise it should be the same as described in the PR